gravitee-apim-e2e/api-test/apis/api-publishing.spec.ts

```diff
@@ -35,18 +50,98 @@ describe('API - Publishing', () => {
         envId,
         newApiEntity: ApisFaker.newApi(),
       });
-      return userApi;
     });
 
     afterAll(async () => {
-      return await apisResourceUser.deleteApi({ orgId, envId, api: userApi.id });
+      await apisResourceUser.deleteApi({ orgId, envId, api: userApi.id });
     });
 
-    describe('As ANONYMOUS user', () => {
-      it('Get APIs should not contain created api', async () => {
+    describe.each`
+      user             | apiResource
+      ${'ANONYMOUS'}   | ${apiResourceAnonymous}
+      ${'ADMIN'}       | ${apiResourceAdmin}
+      ${'API_USER'}    | ${apiResourceApiUser}
+      ${'APP_USER'}    | ${apiResourceAppUser}
+      ${'SIMPLE_USER'} | ${apiResourceSimpleUser}
+    `('As $user user', ({ apiResource }) => {
+      test('Get APIs should not contain created api', async () => {
         const apisResponse = await apiResource.getApis({});
         expect(apisResponse.data.find((api) => api.id === userApi.id)).not.toBeDefined();
       });
+
+      test('Get API by id should return not found', async () => {
+        await fail(
+          apiResource.getApiByApiIdRaw({
+            apiId: userApi.id,
+          }),
+          404,
+        );
+      });
+    });
+  });
+
+  describe('Published', () => {
+    beforeAll(async () => {
+      userApi = await apisResourceUser.createApi({
+        orgId,
+        envId,
+        newApiEntity: ApisFaker.newApi(),
+      });
+      publishedApi = await apisResourceUser.updateApi({
+        api: userApi.id,
+        updateApiEntity: ApisFaker.updateFromApiEntity(userApi, { lifecycle_state: ApiLifecycleState.PUBLISHED }),
+        orgId,
+        envId,
+      });
+    });
+
+    afterAll(async () => {
+      await apisResourceUser.deleteApi({ orgId, envId, api: userApi.id });
+    });
+
+    describe.each`
+      user             | apiResource
+      ${'ANONYMOUS'}   | ${apiResourceAnonymous}
+      ${'ADMIN'}       | ${apiResourceAdmin}
+      ${'APP_USER'}    | ${apiResourceAppUser}
+      ${'SIMPLE_USER'} | ${apiResourceSimpleUser}
+    `('As $user user', ({ apiResource }) => {
+      test('Get APIs should not contain created api', async () => {
+        const apisResponse = await apiResource.getApis({});
+        expect(apisResponse.data.find((api) => api.id === userApi.id)).not.toBeDefined();
+      });
+
+      test('Get API by id should return not found', async () => {
+        await fail(
+          apiResource.getApiByApiIdRaw({
+            apiId: userApi.id,
+          }),
+          404,
+        );
+      });
+    });
+
+    describe('As API_USER user', () => {
+      test('Get APIs should contain created api', async () => {
+        const apisResponse = await apiResourceApiUser.getApis({});
+        expect(apisResponse.data.find((api) => api.id === userApi.id)).toBeDefined();
+      });
+
+      test('Get API by id should return the api', async () => {
+        const response = await apiResourceApiUser.getApiByApiIdRaw({
```

Comment:
Would it be easier to call `getApiByApiId` instead of the `raw` function ?
It returns directly the api object.